aeron-archive/src/main/java/io/aeron/archive/client/ReplayMerge.java

```diff
@@ -175,18 +176,23 @@ public void close()
         final State state = this.state;
         if (State.CLOSED != state)
         {
-            if (!archive.context().aeron().isClosed())
+            final AeronArchive.Context context = archive.context();
+            final Aeron aeron = context.aeron();
+            if (!aeron.isClosed())
             {
+                final ErrorHandler errorHandler = context.errorHandler();
                 if (State.MERGED != state && State.STOP_REPLAY != state)
                 {
-                    subscription.removeDestination(replayDestination);
+                    AeronCloseHelper.close(errorHandler, () -> subscription.removeDestination(replayDestination));
                 }
 
                 if (isReplayActive)
                 {
                     isReplayActive = false;
-                    final long correlationId = archive.context().aeron().nextCorrelationId();
-                    archive.archiveProxy().stopReplay(replaySessionId, correlationId, archive.controlSessionId());
+                    final long correlationId = aeron.nextCorrelationId();
+                    AeronCloseHelper.close(errorHandler,
+                        () -> archive.archiveProxy()
+                        .stopReplay(replaySessionId, correlationId, archive.controlSessionId()));
```

Comment:
This should be dealt with at a higher level in the doWork duty cycle.

Reply:
@mjpt777 There is already a `stopReplay()` which does that and it is called from the `doWork`. Does it mean that the entire ` if (isReplayActive)` block can be removed from the `close()` method?

Reply:
I've pushed a different way to stop the replay which may fail to merge so this should be better.

Reply:
Merged. The close remains as-is from the HEAD.